sig-scalability/slos/dns_programming_latency.md

```diff
@@ -37,6 +39,10 @@ The reason for doing it this way is feasibility for efficiently computing that:
     in 99% of programmers (e.g. iptables). That requires tracking metrics on
     per-change base (which we can't do efficiently).
 
+- The SLI is expected to remain constant independently of the number of records, per
```

Comment:
What's the implication of this?  Scheduler has a finite throughput.  Nodes have finite bandwidth.

If I start a 5 pod headless-service, it's reasonable to expect that DNS for the 5th pod is very very soon after DNS for the 1st:
t0: scale RS to 5
t1: RS controller creates pod 1
t2: scheduler schedules pod 1
t3: kubelet downloads image
t4: kubelet runs pod
t5: runtime assigns an IP
t6: kubelet reports the IP
t7: endpointslice controller observes IP and updates EPSlices
t8: DNS observes EPSlices and updates DNS
```

t1 - t8 happen 5 times, roughly concurrently, and is likely bounded by image download time. 

Change that to 5000 and now you are bounded by the scheduler's throughput.  Is DNS not allowed to publish the 1st IP until the last pod is started?  How does it know which one is last?

Edit:

Or did you mean something like "The time between pod-started-and-IP-assigned and availble-in-DNS should not be significantly different for the 1st vs. last pod" ?  That must be what you meant...

Reply:
>  "The time between pod-started-and-IP-assigned and availble-in-DNS should not be significantly different for the 1st vs. last pod" ? That must be what you meant...


this, if you give me the right sentence in english so this is more clear please add it as a suggestion